hibernate-core/src/main/java/org/hibernate/engine/internal/StatefulPersistenceContext.java

```diff
@@ -131,7 +131,10 @@ the following fields are used in all circumstances, and are not worth (or not su
 	private IdentityHashMap<Object, PersistentCollection<?>> arrayHolders;
 
 	// Identity map of CollectionEntry instances, by the collection wrapper
-	private IdentityMap<PersistentCollection<?>, CollectionEntry> collectionEntries;
+	private InstanceIdentityMap<PersistentCollection<?>, CollectionEntry> collectionEntries;
+
+	// Current collection instance id
+	private int currentCollectionInstanceId;
```

Comment:
```suggestion
	private transient int currentCollectionInstanceId;
```

Reply:
I don't believe we support serialization for `PersistenceContext`, but sure we can do this.

Reply:
We do support serialization :)

Reply:
I've always hated that feature. In fact, I hate it so much I bookmarked this message: https://hibernate.zulipchat.com/#narrow/channel/132096-hibernate-user/topic/EntityEntry.20serialized.20form.20.2F.20type.20.28dis.29assembly